lib/rpush/daemon/fcm/delivery.rb

```diff
@@ -0,0 +1,162 @@
+module Rpush
+  module Daemon
+    module Fcm
+      # https://firebase.google.com/docs/cloud-messaging/server
+      class Delivery < Rpush::Daemon::Delivery
+        include MultiJsonHelper
+
+        HOST = 'https://fcm.googleapis.com'.freeze
+        SCOPE = 'https://www.googleapis.com/auth/firebase.messaging'.freeze
+
+        def initialize(app, http, notification, batch)
+          if necessary_data_exists?(app)
+            @app = app
+            @http = http
+            @notification = notification
+            @batch = batch
+
+            @uri = URI.parse("#{HOST}/v1/projects/#{@app.firebase_project_id || ENV['FIREBASE_PROJECT_ID']}/messages:send")
+          else
+            Rpush.logger.error("Cannot find necessary configuration! Please make sure you have set all necessary ENV variables or firebase_project_id and json_key attributes.")
+          end
+        end
+
+        def perform
+          handle_response(do_post)
+        rescue SocketError => error
+          mark_retryable(@notification, Time.now + 10.seconds, error)
+          raise
+        rescue StandardError => error
+          mark_failed(error)
+          raise
+        ensure
+          @batch.notification_processed
+        end
+
+        protected
+
+        def handle_response(response)
+          case response.code.to_i
```

Comment:
@AnilRh I even added a log here but we're not seeing any logs here it means dispatcher is not even trying sending the fcm v1 notifications on prod env but only few times, what could be the reason?